sdk/ml/azure-ai-ml/azure/ai/ml/_azure_environments.py

```diff
@@ -189,3 +192,130 @@ def _resource_to_scopes(resource):
     """
     scope = resource + "/.default"
     return [scope]
+
+
+_AZUREML_AUTH_CONFIG_DIR_ENV_NAME = 'AZUREML_AUTH_CONFIG_DIR'
+def _get_config_dir():
+    """Folder path for azureml-core to store authentication config"""
+    _AUTH_FOLDER_PATH = os.path.expanduser(os.path.join('~', '.azureml', "auth"))
+    if os.getenv(_AZUREML_AUTH_CONFIG_DIR_ENV_NAME, None):
+        return os.getenv(_AZUREML_AUTH_CONFIG_DIR_ENV_NAME, None)
+    else:
+        if sys.version_info > (3, 0):
+            os.makedirs(_AUTH_FOLDER_PATH, exist_ok=True)
+        else:
+            if not os.path.exists(_AUTH_FOLDER_PATH):
+                os.makedirs(_AUTH_FOLDER_PATH)
+
+        return _AUTH_FOLDER_PATH
+
+GLOBAL_CONFIG_DIR = _get_config_dir()
+CLOUD_CONFIG_FILE = os.path.join(GLOBAL_CONFIG_DIR, 'clouds.config')
+DEFAULT_TIMEOUT = 30
+_DEFAULT_ARM_URL = "https://management.azure.com/metadata/endpoints?api-version=2019-05-01"
+_ARM_METADATA_URL_ENV_NAME = "ARM_METADATA_URL"
```

Comment:
I see variables are defined at multiple places in the file. It would be great to move all the variable at the top of the file.